src/workerd/jsg/ser.c++

```diff
@@ -41,22 +51,73 @@ v8::Maybe<uint32_t> Serializer::GetSharedArrayBufferId(
   return v8::Just(n);
 }
 
+void Serializer::scheduleDataCloneErrorForObject(v8::Isolate* isolate, v8::Local<v8::Object> obj) {
+  jsg::Lock& js = jsg::Lock::from(isolate);
+
+  try {
+    // The default error that V8 would generate is "#<TypeName> could not be cloned." -- for some
+    // reason, it surrounds the type name in "#<>", which seems bizarre? Let's generate a better
+    // error.
+    v8::Local<v8::String> message = js.str(kj::str(
+        "Could not serialize object of type \"", obj->GetConstructorName(), "\". This type does "
+        "not support serialization."));
+    isolate->ThrowException(makeDOMException(isolate, message, "DataCloneError"));
```

Comment:
non-blocking for later: having a `js.throwDomException(...)` would probably be nice.

Reply:
Also, why not just have this defer to `ThrowDataCloneError`?

Reply:
> Also, why not just have this defer to ThrowDataCloneError?

I guess because `ThrowDataCloneError()` contains its own try/catch similar to this function, and does not throw JsExceptionThrown. While calling it would work fine in this case, that's only true because it would be the last thing we call before returning, which seems precarious.